samples/40.timex-resolution/src/main/java/com/microsoft/bot/sample/timex/resolution/Ambiguity.java

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.timex.resolution;
+
+import com.microsoft.recognizers.text.Culture;
+import com.microsoft.recognizers.text.ModelResult;
+import com.microsoft.recognizers.text.datetime.DateTimeRecognizer;
+
+import java.util.Dictionary;
+import java.util.HashSet;
+import java.util.List;
+
+
+   /** TIMEX expressions are designed to represent ambiguous rather than definite dates.
+    * For example:
+    * "Monday" could be any Monday ever.
+    * "May 5th" could be any one of the possible May 5th in the past or the future.
+    * TIMEX does not represent ambiguous times. So if the natural language mentioned 4 o'clock
+    * it could be either 4AM or 4PM. For that the recognizer (and by extension LUIS) would return two TIMEX expressions.
+    * A TIMEX expression can include a date and time parts. So ambiguity of date can be combined with multiple results.
+    * Code that deals with TIMEX expressions is frequently dealing with sets of TIMEX expressions.
+    */
+public class Ambiguity
+{
+    public static void dateAmbiguity()
+    {
+       /**
+        * Run the recognizer.
+        */
+        List<ModelResult> results = DateTimeRecognizer.recognizeDateTime("Either Saturday or Sunday would work.", Culture.English);
+
+        /**
+         * We should find two results in this example.
+         */
+        for (ModelResult result : results)
+        {
+            /**
+             * The resolution includes two example values: going backwards and forwards from NOW in the calendar.
+             */
+            HashSet<String> distinctTimexExpressions = new HashSet<String>();
+            List<Dictionary<String, String>> values = (List<Dictionary<String, String>>)result.resolution.get("values");
+            for (Dictionary<String, String> value : values)
+            {
+                /**
+                 * Each result includes a TIMEX expression that captures the inherent date but not time ambiguity.
+                 * We are interested in the distinct set of TIMEX expressions.
+                 */
+                if (value.get("timex") != null)
+                {
+                    String timex = value.get("timex");
```

Comment:
The problem is that we need to validate if the value "timex" exists before trying to get it, If we do this outside the if I think it will break everytime that the value doesn't exist